### PR TITLE
[mac_frame] simplify ValidatePsdu() function

### PR DESCRIPTION
**Pull request by chshu**
It helps to reduce 176 bytes image size.

### CODE REVIEW

File: src/core/mac/mac_frame.cpp [hunk 1]
```diff
@@ -172,122 +172,11 @@ uint16_t Frame::GetFrameControlField(void) const
 
 otError Frame::ValidatePsdu(void) const
 {
-    otError error = OT_ERROR_PARSE;
-    uint8_t offset = kFcfSize + kDsnSize;
-    uint8_t footerLength = kFcsSize;
-    uint16_t fcf;
-
-    VerifyOrExit((offset + footerLength) <= GetPsduLength());
```
Comment (jwhui) [inline src/core/mac/mac_frame.cpp]: `FindPayloadIndex()` does not include this length check, which I think is important to include.

File: src/core/mac/mac_frame.cpp [hunk 2]
```diff
@@ -172,122 +172,11 @@ uint16_t Frame::GetFrameControlField(void) const
...
-        }
-
-        offset += sizeof(ExtAddress);
-        break;
-
-    default:
-        goto exit;
-    }
-
-    VerifyOrExit((offset + footerLength) <= GetPsduLength());
```
Comment (jwhui) [inline src/core/mac/mac_frame.cpp]: Should include this length check as well.

File: src/core/mac/mac_frame.cpp (hunk 1 above)
Comment (chshu) [reply src/core/mac/mac_frame.cpp]: Added this check to` FindPayloadIndex()`, thanks.

File: src/core/mac/mac_frame.cpp (hunk 2 above)
Comment (chshu) [reply src/core/mac/mac_frame.cpp]: Added this check to` FindPayloadIndex()`, thanks.

**Pull request merged**